src/configuration-requestor.spec.ts

```diff
@@ -423,6 +423,50 @@ describe('ConfigurationRequestor', () => {
             coldStartBanditParameters,
           );
         });
+
+        it('should handle more than 1 cold start', async () => {
+          await configurationRequestor.fetchAndStoreConfigurations();
+          expect(fetchSpy).toHaveBeenCalledTimes(2);
+
+          // setting up scenario where user added new bandit
+          const customResponseMockGenerator = (url: string) => {
+            const responseFile = url.includes('bandits')
+              ? MOCK_BANDIT_MODELS_RESPONSE_FILE
+              : MOCK_FLAGS_WITH_BANDITS_RESPONSE_FILE;
+            const response = readMockUFCResponse(responseFile);
+
+            if ('banditReferences' in response && url.includes('config')) {
+              response.banditReferences.brand_new_bandit = {
+                flagVariations: [
+                  {
+                    key: 'cold_start_bandit',
+                    flagKey: 'cold_start_bandit_flag',
+                    variationKey: 'cold_start_bandit',
+                    variationValue: 'cold_start_bandit',
+                  },
+                ],
+                modelVersion: 'cold start',
+              };
+            } else if (url.includes('bandits') && 'bandits' in response) {
+              response.bandits.brand_new_bandit = {
+                banditKey: 'brand_new_bandit',
+                modelName: 'very new',
+                modelVersion: 'cold start',
+                modelData: {
+                  gamma: 1.0,
+                  defaultActionScore: 0.0,
+                  actionProbabilityFloor: 0.0,
+                  coefficients: {},
+                },
+              };
+            }
+            return response;
+          };
+          initiateFetchSpy(customResponseMockGenerator);
+
+          await configurationRequestor.fetchAndStoreConfigurations();
+          expect(fetchSpy).toHaveBeenCalledTimes(2);
```

Comment:
it's unclear to me what we're verifying here. you have `expect(fetchSpy).toHaveBeenCalledTimes(2);` twice (here and line 429). Are you asserting on no new calls to the `fetchSpy` to have been made?

Reply:
On line 465 fetch spy is reinitialised which resets call count. The test case that I want to cover is for 2 different bandits with same modelVersion(cold start) so I'm expecting 2 calls 1 for ufc 1 for bandits.